Math/NumberTheory/EisensteinIntegers.hs

```diff
@@ -127,4 +131,17 @@ conjugate (a :+ b) = (a - b) :+ (-b)
 
 -- | The square of the magnitude of a Eisenstein integer.
 norm :: EisensteinInteger -> Integer
-norm (a :+ b) = a*a - a * b + b*b
\ No newline at end of file
+norm (a :+ b) = a*a - a * b + b*b
+
+-- | Checks if a given @EisensteinInteger@ is prime. @EisensteinInteger@s
+-- whose norm is a prime congruent to @0@ or @1@ modulo 3 are prime.
+-- See <http://thekeep.eiu.edu/theses/2467 Bandara, Sarada, "An Exposition of the Eisenstein Integers" (2016)>,
+-- page 12.
+isPrime :: EisensteinInteger -> Bool
+isPrime e@(a :+ b)
+    | e == 0                             = False
+    | b == 0 && a `mod` 3 == 2           = Testing.isPrime a
+    | a == 0 && b `mod` 3 == 2           = Testing.isPrime b
+    | nE `mod` 3 == 0 || nE `mod` 3 == 1 = Testing.isPrime nE
```

Comment:
I don't know how I didn't notice what I wrote here, but this is clearly wrong. `nE 'mod' 3 == 0` will never happen for a prime.